### ISSUE TITLE
unknown status in Autosubmit 3.6

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#147]
In GitLab by @fmassonn on Jun 20, 2016, 09:41

Hi,
I have a job that finished successfully on the HPC but was marked as FAILED in the pkl.

End of the autosubmit log file `/home/Earth/fmassonn/logs/autosubmit/log_a07m_20160614140326_bscearth321`:

```
6742 of 11666 jobs remaining (04:03)
Job a07m_19790101_51_ENKF is RUNNING


6742 of 11666 jobs remaining (04:03)
^[[33m[WARNING] Job a07m_19790101_51_ENKF seemed to be completed but there is no COMPLETED file^[[0m^[[39m
^[[33m[WARNING] Job a07m_19790101_51_ENKF in UNKNOWN status^[[0m^[[39m


6742 of 11666 jobs remaining (04:04)
kill: sending signal to 1945095 failed: No such process
^[[33m[WARNING] Job a07m_19790101_51_ENKF seemed to be completed but there is no COMPLETED file^[[0m^[[39m
^[[33mJob a07m_19790101_51_ENKF is FAILED^[[0m^[[39m
No more jobs to run.
Some jobs have failed and reached maximun retrials

```

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Jun 20, 2016, 19:14

In order to research about this issue, can I assume that you are running with paramiko?

Thanks!

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 22, 2016, 10:21

Unfortunately the error appears also with paramiko.

Log file: `/home/Earth/fmassonn/logs/autosubmit/log_a07m_20160622083755_bscearth321`


```
5675 of 11666 jobs remaining (10:12)
[WARNING] Job a07m_19790101_62_ENKF seemed to be completed but there is no COMPLETED file
[WARNING] Job a07m_19790101_62_ENKF in UNKNOWN status


5675 of 11666 jobs remaining (10:14)
kill: sending signal to 1954741 failed: No such process
[WARNING] Job a07m_19790101_62_ENKF seemed to be completed but there is no COMPLETED file
Job a07m_19790101_62_ENKF is FAILED
No more jobs to run.
```


And my configuration file

```
fmassonn@ bscearth321: ~/logs/autosubmit >> tail /esnas/autosubmit/a07m/conf/autosubmit_a07m.conf
# Default = False
NOTIFICATIONS = True
# Mail address where notifications will be received
TO = francois.massonnet@ bsc.es

[communications]
# Communications library used to connect with platforms: paramiko or saga.
# Default = paramiko
API = paramiko
 
```

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jun 22, 2016, 10:39

Hi @fmassonn It seems that the job a07m_19790101_62_ENKF was indeed not completed. See /gpfs/scratch/cns96/cns96177/a07m/LOG_a07m/a07m_19790101_62_ENKF_1954741.err

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 22, 2016, 10:51

Hi, 

That job seems to be completed since the following file exists

`/gpfs/scratch/cns96/cns96177/a07m/LOG_a07m/a07m_19790101_62_ENKF_COMPLETED`

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jun 22, 2016, 11:18

Ah ok! it was still running in MN when I checked sorry.
However Autosubmit failed because the problem comes from the fact that MN returned WNKWN at Jun 22 09:37 (while the job was still running!! O_o) 
The reason why this happens needs to be checked with support... Could you send them an email please ?

See log file `/esnas/autosubmit/a07m/tmp/20160622_083813_run.log`

```
2016-06-22 10:11:52,306 Output JOBID   USER    STAT  QUEUE      FROM_HOST   EXEC_HOST   JOB_NAME   SUBMIT_TIME
1954741 cns9617 RUN   sequential login2      s01         *1_62_ENKF Jun 22 09:37
2016-06-22 10:11:52,306 Job a07m_19790101_62_ENKF is RUNNING
2016-06-22 10:11:52,307 Updating FAILED jobs
2016-06-22 10:11:52,311 Updating WAITING jobs
2016-06-22 10:11:52,320 Update finished
2016-06-22 10:12:02,410 Reloading parameters...
2016-06-22 10:12:02,428 Loading parameters...
2016-06-22 10:12:02,428 Loading project parameters...
2016-06-22 10:12:02,441

5675 of 11666 jobs remaining (10:12)
2016-06-22 10:12:02,441 Sleep: 10
2016-06-22 10:12:02,441 Number of retrials: 1
2016-06-22 10:12:02,930 Command bjobs 1954741 in mn-cns96 successful with out message: JOBID   USER    STAT  QUEUE      FROM_HOST   EXEC_HOST   JOB_NAME   SUBMIT_TIME
1954741 cns9617 UNKWN sequential login2      s01         *1_62_ENKF Jun 22 09:37
2016-06-22 10:12:02,946 Successful check job command: bjobs 1954741
2016-06-22 10:12:02,947 Output JOBID   USER    STAT  QUEUE      FROM_HOST   EXEC_HOST   JOB_NAME   SUBMIT_TIME
1954741 cns9617 UNKWN sequential login2      s01         *1_62_ENKF Jun 22 09:37
2016-06-22 10:12:02,956 Job a07m_19790101_62_ENKF in UNKNOWN status. Checking completed files
```

Explanation of my confusion:
MN tmp folder when I checked the completion of your job at ~ 10:30
```
cns96177@mn1.bsc.es:/gpfs/scratch/cns96/cns96177/a07m/LOG_a07m >> ll a07m_19790101_62_ENKF*
-rw-r--r-- 1 cns96177 cns96 38542 Jun 22 09:40 a07m_19790101_62_ENKF_1954741.err
-rw-r--r-- 1 cns96177 cns96  1405 Jun 22 09:40 a07m_19790101_62_ENKF_1954741.out
-rw-r--r-- 1 cns96177 cns96  9305 Jun 22 09:37 a07m_19790101_62_ENKF.cmd
-rw-r--r-- 1 cns96177 cns96    11 Jun 22 09:38 a07m_19790101_62_ENKF_STAT
```
MN tmp folder when you repliead me that the job was completed at ~ 10:50
```
cns96177@mn1.bsc.es:/gpfs/scratch/cns96/cns96177/a07m/LOG_a07m >> ll a07m_19790101_62_ENKF*
-rw-r--r-- 1 cns96177 cns96   59297 Jun 22 10:52 a07m_19790101_62_ENKF_1954741.err
-rw-r--r-- 1 cns96177 cns96 7975603 Jun 22 10:52 a07m_19790101_62_ENKF_1954741.out
-rw-r--r-- 1 cns96177 cns96    9305 Jun 22 09:37 a07m_19790101_62_ENKF.cmd
-rw-r--r-- 1 cns96177 cns96       0 Jun 22 10:52 a07m_19790101_62_ENKF_COMPLETED
-rw-r--r-- 1 cns96177 cns96      22 Jun 22 10:52 a07m_19790101_62_ENKF_STAT
```

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 22, 2016, 11:57

Thanks Mingu,

I have asked support by CCing you.

François

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 22, 2016, 13:48

The message from Support:


Dear François,

I have no access to /esnas so I cannot see the contents of your log file. Also, you have 3 users and didn't specify which user is the one with the problem of this job.

The Unknown state is caused because the master LSF program cannot contact with the execution node at a certain time. This can be caused by network or hardware issues or administration activity and is unavoidable from a user's point of view. The only real solution is that your code accounts for the UNKNOWN state and fails gracefully or handles it appropriately.

Regards,
Miguel

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jun 22, 2016, 17:21

hmmm ok we are accounting for the UNKNOWN stat and Autosubmit fails gracefully... However I need to check because I had in mind that a retrial was introduced (see #92) to handle it more appropriately than failing gracefully :speak_no_evil:

**Comment by kinow** [comment]
In GitLab by @obellprat on Jun 23, 2016, 13:20

I just had the same issue running with autosubmit -v 3.6.0 , just to report it.

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 23, 2016, 15:06

Yes, one more for me:

log: `~fmassonn/logs/autosubmit/log_a07m_20160623092258_bscearth321`

```
5287 of 11666 jobs remaining (12:52)
[WARNING] Job a07m_19790101_66_ENKF seemed to be completed but there is no COMPLETED file
[WARNING] Job a07m_19790101_66_ENKF in UNKNOWN status


5287 of 11666 jobs remaining (12:53)
kill: sending signal to 1958432 failed: No such process
[WARNING] Job a07m_19790101_66_ENKF seemed to be completed but there is no COMPLETED file
Job a07m_19790101_66_ENKF is FAILED
No more jobs to run.
Some jobs have failed and reached maximun retrials

```

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 23, 2016, 15:32

And another one:
(Note: I'm continuing reporting so that maybe we see similarity between the type of jobs that produce this error)

`~fmassonn/logs/autosubmit/log_a082_20160622175010_bscearth321`

```
9652 of 11666 jobs remaining (15:27)
[WARNING] Job a082_19840101_21_ENKF seemed to be completed but there is no COMPLETED file
[WARNING] Job a082_19840101_21_ENKF in UNKNOWN status


9652 of 11666 jobs remaining (15:29)
kill: sending signal to 1959239 failed: No such process
[WARNING] Job a082_19840101_21_ENKF seemed to be completed but there is no COMPLETED file
Job a082_19840101_21_ENKF is FAILED
No more jobs to run.
Some jobs have failed and reached maximun retrials

```

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jun 23, 2016, 18:41

Hi @fmassonn and @obellprat.
I think I found a fix for this ... Could you try using the development branch that also try to solve memory problems ?
The steps to try the development branch is described here https://earth.bsc.es/gitlab/es/autosubmit/issues/148#note_3221

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 23, 2016, 18:46

I'm trying this for the next submissions.

Thanks anyway for your help @dmanubens and happy Sant Joan!

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 29, 2016, 15:49

I'm testing now the procedure with experiment `a077` that failed in the same way as described above:
```
fmassonn@ bscearth321: ~/logs/autosubmit >> cat /home/Earth/fmassonn/logs/autosubmit/log_a077_20160624135651_bscearth321 | grep seemed
[WARNING] Job a077_19580101_fc00_81_CLEAN seemed to be completed but there is no COMPLETED file
[WARNING] Job a077_19580101_fc00_81_CLEAN seemed to be completed but there is no COMPLETED file
```

More to follow.

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 29, 2016, 15:56

Here is what I did:

```
module unload autosubmit 
module unload Python
mkdir -p ~/venvs/as_361/
virtualenv ~/venvs/as_361
source ~/venvs/as_361/bin/activate
pip install https://earth.bsc.es/gitlab/es/autosubmit/repository/archive.zip?ref=develop-pkl-2-db --upgrade --no-deps
```

(The last thing as suggested by you in issue #148 )

But I get this error when running autosubmit -v:

```
(as_361)fmassonn@ bscearth321: ~ >> autosubmit -v
Traceback (most recent call last):
  File "/home/Earth/fmassonn/venvs/as_361/bin/autosubmit", line 29, in <module>
    from autosubmit.autosubmit import Autosubmit
  File "/home/Earth/fmassonn/venvs/as_361/lib/python2.7/site-packages/autosubmit/autosubmit.py", line 46, in <module>
    import dialog
ImportError: No module named dialog
```

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jun 29, 2016, 16:06

Hi @fmassonn , if it is the first time you create the virtualenv remove the `--upgrade --no-deps` option:
```
pip install https://earth.bsc.es/gitlab/es/autosubmit/repository/archive.zip?ref=develop-pkl-2-db
```
Cheers

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 29, 2016, 16:10

OK thanks, I was deleting venvs each time :-/

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 29, 2016, 16:15

Hey @dmanubens 
Unfortunately this still does not work:

```
rm -rf ~/venvs/; module unload autosubmit; module unload Python; mkdir -p ~/venvs/as_361/; virtualenv ~/venvs/as_361 ; source ~/venvs/as_361/bin/activate ; pip install https://earth.bsc.es/gitlab/es/autosubmit/repository/archive.zip?ref=develop-pkl-2-db
```

I have saved the log of the last command in 
`/home/Earth/fmassonn/log_pip`

And "autosubmit -v " does not work
```
(as_361)fmassonn@ bscearth321: ~ >> autosubmit -v
Traceback (most recent call last):
  File "/shared/earth/software/bin/autosubmit", line 29, in <module>
    from autosubmit.autosubmit import Autosubmit
ImportError: No module named autosubmit.autosubmit

```

François

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jun 29, 2016, 19:31

Hey @fmassonn could it be that you need to do a `module purge` instead of `module unload autosubmit; module unload Python;` ?

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 30, 2016, 09:15

```
rm -rf ~/venvs/; module purge; mkdir -p ~/venvs/as_361/; virtualenv ~/venvs/as_361 ; source ~/venvs/as_361/bin/activate ; pip install https://earth.bsc.es/gitlab/es/autosubmit/repository/archive.zip?ref=develop-pkl-2-db
```

[WORKS]

```
cp /esnas/autosubmit/a077/pkl/job_list_a077.pkl /esnas/autosubmit/a077/pkl/job_list_a077.pkl.bkp
autosubmit create a077
autosubmit recovery -s -all a077
expid=a077; nohup autosubmit run ${expid} >& ~/logs/autosubmit/log_${expid}_`date +%Y%m%d%H%M%S`_${HOST} &
```

[WORKS]. Job is now submitted. Logs are in `/home/Earth/fmassonn/logs/autosubmit/log_a077_20160630091739_bscearth321`

Thanks @dmanubens for valuable help, I'll let you know if I have trouble.

François

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jul 1, 2016, 10:42

Dear @dmanubens and @ jlope2 

So far so good, I've been running `a077` with autosubmit v3.6.1 since 24h and I haven't encountered any problem.

That said, before making hasty conclusions, I will test the branch on another experiment that involves many members (`a07m`). If I do not have problems by monday, I guess you can release 3.6.1 -- at least from my side.

François

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jul 1, 2016, 13:57

Hi,

I am unable to monitor `a07m` within the VirtualEnv:

```
(as_361)fmassonn@ bscearth321: ~/logs/autosubmit >> autosubmit monitor -o png -fs FAILED a07m
Getting job list...

Checking configuration files...
autosubmit_a07m.conf OK
platforms_a07m.conf OK
jobs_a07m.conf OK
expdef_a07m.conf  OK
Configuration files OK

Creating jobs...
Loading JobList
[CRITICAL] Unhandled exception on Autosubmit: iteration over non-sequence
Traceback (most recent call last):
  File "/home/Earth/fmassonn/venvs/as_361/lib/python2.7/site-packages/autosubmit/autosubmit.py", line 297, in parse_args
    args.filter_type, args.hide)
  File "/home/Earth/fmassonn/venvs/as_361/lib/python2.7/site-packages/autosubmit/autosubmit.py", line 733, in monitor
    job_list = Autosubmit.load_job_list(expid, as_conf)
  File "/home/Earth/fmassonn/venvs/as_361/lib/python2.7/site-packages/autosubmit/autosubmit.py", line 2026, in load_job_list
    date_format, as_conf.get_retrials(), False)
  File "/home/Earth/fmassonn/venvs/as_361/lib/python2.7/site-packages/autosubmit/job/job_list.py", line 109, in generate
    jobs_data = {str(row[0]): row for row in self.load()}
TypeError: iteration over non-sequence

```

François

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Jul 1, 2016, 15:18

Hi @fmassonn,

I think that the pickle (.pkl) that you are trying to load was generated with an old version.

If you want to make the monitor with this, the development version, you have to re-create the experiment with it.

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jul 1, 2016, 15:20

Hi,

But that's not possible: I saved the old pickle (generated with 3.6.0), then re-created the experiment with 3.6.1, and recovered it. Shall I re-create it anyway?

François

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Jul 1, 2016, 15:33

Probably you re-created the experiment with the development version one week ago, however, as it is a development version, we are doing some changes everyday, so now, a recreation is needed.

Thanks,

PD: Only a clarification: the autosubmit 3.6.x is the version which is already on production (available). The next version will be named 3.7.x and above (which we are developing now)

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Jul 8, 2016, 16:37

Everything was OK?

I think that the original issue was solved so I proceed to close this issue.

If something else happens, open a new ticket (issue), please.

Thanks!

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jul 13, 2016, 11:11

Hi @ jlope2 

I'm not sure that the issue is fixed. I was running `a077` from within the virtual environment and still get the same error:

```
fmassonn@ bscearth321: ~/logs/autosubmit >> grep -A 2 -B 2 "seemed" /home/Earth/fmassonn/logs/autosubmit/log_a077_20160630091739_bscearth321
1670 of 2019 jobs remaining (16:08)
Job a077_19580101_fc00_88_SIM is RUNNING
[WARNING] Job a077_19580101_fc00_87_CLEAN seemed to be completed but there is no COMPLETED file
[WARNING] Job a077_19580101_fc00_87_CLEAN in UNKNOWN status

--

1650 of 2019 jobs remaining (12:34)
[WARNING] Job a077_19580101_fc00_92_CLEAN seemed to be completed but there is no COMPLETED file
[WARNING] Job a077_19580101_fc00_92_CLEAN in UNKNOWN status
Job a077_19580101_fc00_93_SIM is QUEUING
--

1646 of 2019 jobs remaining (14:19)
[WARNING] Job a077_19580101_fc00_93_CLEAN seemed to be completed but there is no COMPLETED file
[WARNING] Job a077_19580101_fc00_93_CLEAN in UNKNOWN status
Job a077_19580101_fc00_94_SIM is QUEUING
--
1634 of 2019 jobs remaining (06:53)
Job a077_19580101_fc00_97_SIM is RUNNING
[WARNING] Job a077_19580101_fc00_96_CLEAN seemed to be completed but there is no COMPLETED file
[WARNING] Job a077_19580101_fc00_96_CLEAN in UNKNOWN status
```

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Jul 13, 2016, 11:48

Okay,

My suggestion is to remain the issue open until we release the new version (this week).
Then you can check if the problem persists or not.

Thanks!

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jul 26, 2016, 11:18

Hi @fmassonn in fact you see those messages in the log and they are normal. We are accounting for the UNKNOWN stat (weird state that the LSF scheduler reports wrongly) therefore the logs are capturing it. However, now Autosubmit handles it appropriately by re-checking the status. Second time the job is checked, in most of cases LSF reports it is in RUN state (real state). Could you check that those jobs you mention finally COMPLETED without your intervention ? Thanks for your help.

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jul 26, 2016, 11:49

Hi @dmanubens , yes, usually the jobs finally complete (a file *_COMPLETED is present on the HPC). 

François

**Comment by kinow** [comment]
In GitLab by @dmanubens on Jul 28, 2016, 12:10

Status changed to closed by commit 700076115a23d048d936fa650bc2be9e53258830
